nixos/modules/services/databases/postgresql.md

```diff
@@ -205,12 +206,12 @@ For an upgrade, a script like this can be used to simplify the process:
 
       export NEWBIN="${newPostgres}/bin"
 
-      export OLDDATA="${config.services.postgresql.dataDir}"
-      export OLDBIN="${config.services.postgresql.package}/bin"
+      export OLDDATA="${cfg.dataDir}"
+      export OLDBIN="${cfg.package}/bin"
 
       install -d -m 0700 -o postgres -g postgres "$NEWDATA"
       cd "$NEWDATA"
-      sudo -u postgres $NEWBIN/initdb -D "$NEWDATA"
+      sudo -u postgres $NEWBIN/initdb -D "$NEWDATA" ${builtins.concatStringsSep " " cfg.initdbArgs}
```

Comment:
I'm a little surprised this isn't `escapeShellArgs`, but since it's the same in the module, I'll leave it as-is. Will fix it up in a follow-up PR.

Reply:
https://github.com/NixOS/nixpkgs/pull/345341